pkg/util/hvutil/hvutil.go

```diff
@@ -23,3 +24,33 @@ func GetPodNamespace(annotations map[string]string) string {
 
 	return annotations[cri.SandboxNamespace]
 }
+
+func sanitize(input string) string {
+
+	var output string
+
+	for _, c := range strings.ToLower(input) {
+		if !(('a' <= c && c <= 'z') || ('0' <= c && c <= '9') || c == '-') {
+			c = '-'
+		}
+		output += string(c)
+	}
+
+	return output
+}
+
+func GetInstanceName(nodeName, podNamespace, podName, sandboxID string) string {
+
+	nodeName = sanitize(nodeName)
+	podNamespace = sanitize(podNamespace)
+	podName = sanitize(podName)
+	sandboxID = sanitize(sandboxID)
+
+	vmName := fmt.Sprintf("%s-%s-%s-%.8s", nodeName, podNamespace, podName, sandboxID)
+
+	if c := vmName[0]; !('a' <= c && c <= 'z') {
+		vmName = "ip-" + vmName
```

Comment:
nit: how about using a prefix unique to caa (eg. `podvm` or `caa-vm-`) to make it easier for querying ?
Also should this method also restrict the name to a specified length ?

Reply:
OK. I'll use prefix 'podvm`.